vendor/k8s.io/kubernetes/pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -259,40 +260,21 @@ func configFromEnv() (cfg Config, ok bool) {
 	return
 }
 
-func createKubernetesClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
-	klog.Info("Creating kubernetes API client.")
-
-	cfg, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
-	if err != nil {
-		return nil, err
-	}
-
-	client, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		return nil, err
-	}
-
-	v, err := client.Discovery().ServerVersion()
-	if err != nil {
-		return nil, err
-	}
-
-	klog.Infof("Kubernetes API client created, server version %s", fmt.Sprintf("v%v.%v", v.Major, v.Minor))
-	return client, nil
+// SetInformers initializes secret lister
+func (os *OpenStack) SetInformers(informerFactory informers.SharedInformerFactory) {
+	os.secretLister = informerFactory.Core().V1().Secrets().Lister()
```

Comment:
Is this implementing some interface? Add a type assertion.

Reply:
Is this called in all contexts?

Reply:
What type assertion do you mean here?